src/components/editor/EditorLogin.jsx

```diff
@@ -51,6 +47,42 @@ export default class Login extends BaseComponent {
     }
   }
 
+  handleRedirect() {
+    let url = this.props.location.query.url || '';
+    if (url) {
+      _.forEach(this.props.location.query, (q, name) => {
+        if (name !== 'url') {
```

Comment:
I know this is my code, but as I was reading it during my last review when I told you to rewrite it I was actually confused for a second what it did until I recalled it, which means this needs a comment. Could you maybe just add a comment above this loop that says something like `// Add additional query params to new url` or maybe it could be phrased even clearer than that...